pkg/restapi/operation/common/models.go

```diff
@@ -42,3 +42,63 @@ type CreatePublicDIDResponse struct {
 	// TODO return base64-encoded raw bytes of the DID doc [Issue: #855]
 	DID *did.Doc `json:"did"`
 }
+
+// RegisterMessageServiceRequest model
+//
+// This is used for operation to register a message service to message handler
+//
+// swagger:parameters registerMsgSvc
+type RegisterMessageServiceRequest struct {
+	// Params for registering message service
+	//
+	// in: body
+	Params *RegisterMsgSvcParams
+}
+
+// RegisterMsgSvcParams contains parameters for registering a message service to message handler
+type RegisterMsgSvcParams struct {
+	// Name of the message service
+	// required: true
+	Name string `json:"name"`
+
+	// Acceptance criteria for message service based on message purpose
+	// in case of multiple purposes, message will be dispatched if any one of the purpose matches
+	// with the purpose of incoming message.
+	// Can be provided in conjunction with other acceptance criteria.
+	Purpose []string `json:"purpose"`
```

Comment:
@sudeshrshetty @Moopli @llorllale I think Filip is updating the design to no longer need purpose?

Reply:
Will talk to Filip and create an issue for it.

Reply:
I'm not sure which is better - keeping vs removing. I think it's better to keep, since it means each protocol can use the same filtering mechanism.